src/codecatalyst/commands.ts

```diff
@@ -135,13 +136,43 @@ function createClientInjector(authProvider: CodeCatalystAuthenticationProvider):
 
         await authProvider.restore()
         const conn = authProvider.activeConnection ?? (await authProvider.promptNotConnected())
-        const client = await createClient(conn)
+        const authenticatedConn = await authenticateConnection(conn, authProvider.auth)
+        const client = await createClient(authenticatedConn)
         telemetry.record({ userId: client.identity.id })
 
         return command(client, ...args)
     }
 }
 
+/**
+ * Returns a connection that is ensured to be authenticated.
+ *
+ * Provides the user the ability to re-authenticate if needed,
+ * otherwise throwing an error.
+ */
+async function authenticateConnection(conn: SsoConnection, auth: Auth): Promise<SsoConnection> {
```

Comment:
elsewhere we have called this "validate connection", is it intentional that we are using different jargon here? i.e. are the semantics different?
```suggestion
async function validateConnection(conn: SsoConnection, auth: Auth): Promise<SsoConnection> {
```

Reply:
From my understanding we have a 'valid' connection since we know it has valid properties (eg: startUrl), but due to the connection expiring we need to authenticate the existing connection with the server.

Successfully authenticating ourselves implies our connection properties were valid??

Reply:
I've been using "validation" when checking local state/configuration whereas "authentication" involves API calls. Right now there's no "who am I" API call for bearer tokens so the word "authenticate" is used only for the browser login flow. There's obviously a lot of overlap though `validateConnection` is more aligned with the impl. in `auth.ts`